v1/mutate/mutate.go

```diff
@@ -79,10 +79,22 @@ func Append(base v1.Image, adds ...Addendum) (v1.Image, error) {
 	history := image.configFile.History
 
 	for _, add := range adds {
-		diffID, err := add.Layer.DiffID()
+		digest, err := add.Layer.Digest()
 		if err != nil {
 			return nil, err
 		}
+		var diffID v1.Hash
+		if add.Image != nil {
+			diffID, err = partial.BlobToDiffID(add.Image, digest)
+			if err != nil {
+				return nil, err
+			}
+		} else {
+			diffID, err = add.Layer.DiffID()
+			if err != nil {
+				return nil, err
+			}
+		}
```

Comment:
I'm fine with this as a band-aid, but I'd like to make Layer.DiffID elide the Blob fetch whenever possible.

Reply:
Can you open an issue to track that?  It was something I'd meant to do, but fell off my radar.